File: frontends/main/src/app/configurator/services/api/cron.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {Cron} from '@core/model/cron';

@Injectable({
	providedIn: 'root'
})
export class CronService {
	constructor(private http: HttpClient) {}

	getCrons(projectId: string): Observable<Cron[]> {
		return this.http.get<Cron[]>(`/api/superuser/configurator/projects/${projectId}/config/crons`);
	}

	getCron(projectId: string, cronId: string): Observable<Cron> {
		return this.http.get<Cron>(`/api/superuser/configurator/projects/${projectId}/config/crons/${cronId}`);
	}

	createCron(projectId: string, cron: Cron): Observable<Cron> {
		return this.http.post<Cron>(`/api/superuser/configurator/projects/${projectId}/config/crons`, cron);
	}

	updateCron(projectId: string, cronId: string, cron: Cron): Observable<Cron> {
		return this.http.put<Cron>(`/api/superuser/configurator/projects/${projectId}/config/crons/${cronId}`, cron);
	}

	deleteCron(projectId: string, cronId: string): Observable<void> {
		return this.http.delete<void>(`/api/superuser/configurator/projects/${projectId}/config/crons/${cronId}`);
	}
}

File: backend/src/main/java/ch/rodano/core/services/bll/configurator/CronService.java
package ch.rodano.core.services.bll.configurator;

import java.util.List;
import java.util.UUID;

import ch.rodano.api.config.CronDTO;

public interface CronService {

	List<CronDTO> getCrons(UUID projectId);

	CronDTO getCron(UUID projectId, UUID cronId);

	CronDTO createCron(UUID projectId, CronDTO dto);

	CronDTO updateCron(UUID projectId, UUID cronId, CronDTO dto);

	void deleteCron(UUID projectId, UUID cronId);
}
